macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,275 @@
+meta:
+  id: mac_os_resource_snd
+  title: Classic MacOS Sound Resource
+  application: Sound Manager
+  endian: be
+  license: MIT
+  xref:
+    wikidata: Q7564684
+doc-ref: "https://developer.apple.com/library/archive/documentation/mac/pdf/Sound/Sound_Manager.pdf"
+seq:
+  - id: format
+    type: u2
+  - id: num_data_formats
+    -orig-id: number_of_data_formats
+    type: u2
+    if: format == 1
+  - id: data_formats
+    type: data_format
+    repeat: expr
+    repeat-expr: num_data_formats
+    if: format == 1
+  - id: reference_count
+    type: u2
+    if: format == 2
+  - id: num_sound_commands
+    -orig-id: number_of_sound_commands
+    type: u2
+  - id: sound_commands
+    type: sound_command
+    repeat: expr
+    repeat-expr: num_sound_commands
+types:
+  data_format:
+    seq:
+      - id: id
+        type: u2
+        enum: data_type
+      - id: options
+        type: u4
+  sound_command:
+    seq:
+      - id: is_data_offset
+        type: b1
+      - id: cmd
+        type: b15
+        enum: cmd_type
+      - id: param1
+        type: u2
+      - id: param2
+        type: u4
+    instances:
+      sound_header_type:
+        pos: param2 + 20
+        type: u1
+        enum: sound_header_type
+        if: is_data_offset and cmd == cmd_type::buffer_cmd
+      sound_header:
+        pos: param2
+        size-eos: true
+        if: is_data_offset and cmd == cmd_type::buffer_cmd
+        type:
+          switch-on: sound_header_type
+          cases:
+            sound_header_type::standard: standard_sound_header
+            sound_header_type::extended: extended_sound_header
+            sound_header_type::compressed: compressed_sound_header
+  standard_sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or NIL if samples follow data structure)
+      - id: length
+        type: u4
+        doc: number of samples in array
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: u2
+        doc: sample rate (Fixed Point, Integer part)
+      - id: sample_rate_frac
+        -orig-id: sampleRate
+        type: u2
+        doc: sample rate (Fixed Point, Fraction part)
```

Comment:
I see that you're using this couple of fields 3 times in the spec. Can you extract them to a new user type, which will additionally calculate the final value of _sample rate_? I mean putting the _Integer part_ and _Fraction part_ of the fixed point number together using a [value instance](https://doc.kaitai.io/user_guide.html#_calculated_value_instances).